cylc/flow/scheduler.py

```diff
@@ -1636,11 +1648,36 @@ async def main_loop(self) -> None:
 
             # Update state summary, database, and uifeed
             self.workflow_db_mgr.put_task_event_timers(self.task_events_mgr)
-            has_updated = await self.update_data_structure()
-            if has_updated and not self.is_stalled:
-                # Stop the stalled timer.
+
+            # List of task whose states have changed.
+            updated_task_list = [
+                t for t in self.pool.get_tasks() if t.state.is_updated]
+            has_updated = updated_task_list or self.is_updated
+
+            if updated_task_list and self.is_restart_timeout_wait:
+                # Stop restart timeout if action has been triggered.
                 with suppress(KeyError):
-                    self.timers[self.EVENT_STALL_TIMEOUT].stop()
+                    self.timers[self.EVENT_RESTART_TIMEOUT].stop()
+                    self.is_restart_timeout_wait = False
+
+            if has_updated:
+                # Update the datastore.
+                await self.update_data_structure(self.is_reloaded)
+
+                if not self.is_reloaded:
+                    # (A reload cannot unstall workflow by itself)
```

Comment:
```suggestion
                    # (A reload cannot install workflow by itself)
```

Reply:
unstall is correct

Reply:
```suggestion
                    # (A reload cannot un-stall workflow by itself)
```

Reply:
I find @MetRonnie 's hyphenation useful here.